### PR TITLE
Cirrus: Follow-up after initial implementation

### PR DESCRIPTION
**Pull request by ldoktor**
Add some description that might speed-up the future reviews and expand the RHEL CI coverage by mocking the required deps.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -16,6 +16,10 @@ fedora_30_task:
     matrix:
       - SETUP: setup.py develop --user
       - SETUP: -m pip install .
+  # Latest pip available avocado-71.0 contains a bug
+  # Latest pip available 69.1 contains a bug
+  allow_failures: $AVOCADO_SRC == '' || $AVOCADO_SRC == 'avocado-framework<70.0'
+  skip_notifications: $AVOCADO_SRC == '' || $AVOCADO_SRC == 'avocado-framework<70.0'
```
Comment (ldoktor) [inline .cirrus.yml]: For some reason the "allow_failures" still propagated to github, therefor I added the "skip_notification" which took care of it. Hopefully we'll publish the new LTS soon to get rid of one skip and new Avocado 72.0 (or minor 71.1?) to remove the other failure.

File: .cirrus.yml [hunk 2]
```diff
@@ -16,6 +16,10 @@ fedora_30_task:
     matrix:
       - SETUP: setup.py develop --user
       - SETUP: -m pip install .
+  # Latest pip available avocado-71.0 contains a bug
```
Comment (clebergnu) [inline .cirrus.yml]: Which bug?

Comment (ldoktor) [reply .cirrus.yml]: This one https://github.com/avocado-framework/avocado/pull/3309

File: .cirrus.yml [hunk 3]
```diff
@@ -44,10 +47,20 @@ rhel_8_task:
...
+  # Latest pip available 69.1 contains a bug
+  allow_failures: $AVOCADO_SRC == '' || $AVOCADO_SRC == 'avocado-framework<70.0'
+  skip_notifications: $AVOCADO_SRC == '' || $AVOCADO_SRC == 'avocado-framework<70.0'
   setup_script:
     *setup_scr
-  plugin_script:
-    - $PYTHON -m avocado plugins 2>/dev/null | grep vt-bootstrap
+  mock_deps_script:
+    - for PROG in tcpdump qemu-kvm qemu-img qemu-io; do touch /usr/local/bin/$PROG; chmod +x /usr/local/bin/$PROG; done
```
Comment (clebergnu) [inline .cirrus.yml]: If we choose to fake the programs, we can do that just once in the container images.

File: .cirrus.yml (hunk 1 above)
Comment (clebergnu) [reply .cirrus.yml]: With the newly released 69.2, there's no need anymore for those.

File: .cirrus.yml [hunk 4]
```diff
@@ -7,8 +7,11 @@ fedora_30_task:
   env:
     PYTHON: python3
     matrix:
+      # Get deps from setup.py (therefor pip version of avocado)
```
Comment (clebergnu) [inline .cirrus.yml]: Fun fact: my dictionary says that "therefor" is an "archaic" word, while "therefore" is the "modern" one.

Comment (ldoktor) [reply .cirrus.yml]: Therefor is 1 char shorter, therefor it's better for the environment (-108 chars to describe the benefit :D)

File: .cirrus.yml (hunk 3 above)
Comment (ldoktor) [reply .cirrus.yml]: good point

File: .cirrus.yml (hunk 1 above)
Comment (ldoktor) [reply .cirrus.yml]: OK, removing.

**Comment by ldoktor** [comment]
@clebergnu addressed the issues, would you please re-create the docker image based on this dockerfile to get the CI to pass?

**Comment by ldoktor** [comment]
Oh, 72 was also deployed already, so removed all skips, the only issue should be the outdated container image for el8.

File: .cirrus.yml (hunk 4 above)
Comment (clebergnu) [reply .cirrus.yml]: :D

**Comment by clebergnu** [comment]
@ldoktor image f56980be40ad built.

**Comment by ldoktor** [comment]
@clebergnu re-triggered and passed

**Comment by ldoktor** [comment]
I guess it's fine to merge this with 1 ack since it's CI script and no actual code change...

**Comment by clebergnu** [comment]
@ldoktor yep, merging it.

**Pull request merged**